### PR TITLE
Add an authentication check & page to request access

### PR DESCRIPTION
**Pull request by OrdiNeu**
## Ticket(s)

- [DIG-1857](https://candig.atlassian.net/browse/DIG-1857)

## Description 

- This allows a logged-in user to see a request access page if they are not yet in OPA, and to request access.

# Screenshots
image

image

## To do/Tickets to be made before merging branch

- This should be tested via https://github.com/CanDIG/CanDIGv2/pull/914

## Types of Change(s)

-   [ ] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [x] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [ ] My change requires a change to the documentation
-   [ ] I have updated the documentation accordingly
-   [ ] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [ ] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [ ] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots

[DIG-1857]: https://candig.atlassian.net/browse/DIG-1857?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/pages/authentication/AuthDisplay.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                    <>
+                        <Typography variant="h2">
+                            You are not authorized to access this site and your authorization request is still pending.
+                        </Typography>
+                        <Typography variant="h2">Please contact {config.supportEmail} for details</Typography>
+                    </>
+                }
+                button={
+                    <Button variant="contained" href={`mailto: ${config.supportEmail}`}>
+                        Contact
```
Comment (CourtneyGosselin) [inline src/views/pages/authentication/AuthDisplay.js]: If a support email isn’t set, should we remove the contact button and advise users to reach out to their admins/your site?

File: public/index.html [hunk 2]
```diff
@@ -28,7 +28,7 @@
     </head>
 
     <body>
-        <noscript>You need to enable JavaScript to run this app.</noscript>
+        <noscript>You need to enable JavaScript to run this app. <link rel="test" href="%PUBLIC_URL%/iframetest.html" /></noscript>
```
Comment (CourtneyGosselin) [inline public/index.html]: Did you mean to push this?

**Review by CourtneyGosselin — CHANGES_REQUESTED** [review]
I’d suggest a few minor visual changes. The shadow on the entire dialog is missing, as well as the shadow on the gray container, but these aren’t necessary if we prefer a flatter design. I’d change the pending hourglass to yellow, add 2–3em of spacing above the gray container and below the contact/request button, and adjust the hourglass padding to around 1em above and below. Also in the design the buttons were rounded to match other buttons across the site. Finally, the dialog box should be centered within the blue main container.

I also see some linter errors some which will resolve by merging in the develop branch
image

File: public/index.html (hunk 2 above)
Comment (OrdiNeu) [reply public/index.html]: My apologies, that was for testing something unrelated

**Review by CourtneyGosselin — APPROVED** [review]
LGTM just some minor linter warnings tested with user2

**Pull request merged**